tests/components/blebox/test_light.py

```diff
@@ -0,0 +1,597 @@
+    assert state.state == STATE_ON
+
+
+async def test_dimmer_off(dimmer, hass, config):
+    """Test light off."""
+
+    feature_mock, entity_id = dimmer
+
+    def initial_update():
+        feature_mock.is_on = True
+
+    feature_mock.async_update = AsyncMock(side_effect=initial_update)
+    await async_setup_entity(hass, config, entity_id)
+    feature_mock.async_update = AsyncMock()
+
+    state = hass.states.get(entity_id)
+    assert state.state == STATE_ON
+
+    def turn_off():
+        feature_mock.is_on = False
+        feature_mock.brightness = 0  # off
+
+    feature_mock.async_off = AsyncMock(side_effect=turn_off)
+    await hass.services.async_call(
+        "light", SERVICE_TURN_OFF, {"entity_id": entity_id}, blocking=True,
+    )
+
+    state = hass.states.get(entity_id)
+    assert state.state == STATE_OFF
+    assert ATTR_BRIGHTNESS not in state.attributes
+
+
+@pytest.fixture(name="wlightbox_s")
+def wlightboxs_fixture():
+    """Return a default light entity mock."""
+
+    feature = mock_feature(
+        "lights",
+        blebox_uniapi.light.Light,
+        unique_id="BleBox-wLightBoxS-1afe34e750b8-color",
+        full_name="wLightBoxS-color",
+        device_class=None,
+        brightness=None,
+        is_on=None,
+        supports_color=False,
+        supports_white=False,
+    )
+    product = feature.product
+    type(product).name = PropertyMock(return_value="My wLightBoxS")
+    type(product).model = PropertyMock(return_value="wLightBoxS")
+    return (feature, "light.wlightboxs_color")
+
+
+async def test_wlightbox_s_init(wlightbox_s, hass, config):
+    """Test cover default state."""
+
+    _, entity_id = wlightbox_s
+    entry = await async_setup_entity(hass, config, entity_id)
+    assert entry.unique_id == "BleBox-wLightBoxS-1afe34e750b8-color"
+
+    state = hass.states.get(entity_id)
+    assert state.name == "wLightBoxS-color"
+
+    supported_features = state.attributes[ATTR_SUPPORTED_FEATURES]
+    assert supported_features & SUPPORT_BRIGHTNESS
+
+    assert ATTR_BRIGHTNESS not in state.attributes
+    assert state.state == STATE_OFF
+
+    device_registry = await hass.helpers.device_registry.async_get_registry()
+    device = device_registry.async_get(entry.device_id)
+
+    assert device.name == "My wLightBoxS"
+    assert device.identifiers == {("blebox", "abcd0123ef5678")}
+    assert device.manufacturer == "BleBox"
+    assert device.model == "wLightBoxS"
+    assert device.sw_version == "1.23"
+
+
+async def test_wlightbox_s_update(wlightbox_s, hass, config):
+    """Test light updating."""
+
+    feature_mock, entity_id = wlightbox_s
+
+    def initial_update():
+        feature_mock.brightness = 0xAB
+        feature_mock.is_on = True
+
+    feature_mock.async_update = AsyncMock(side_effect=initial_update)
+
+    await async_setup_entity(hass, config, entity_id)
+
+    state = hass.states.get(entity_id)
+    assert state.state == STATE_ON
+    assert state.attributes[ATTR_BRIGHTNESS] == 0xAB
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5535891)